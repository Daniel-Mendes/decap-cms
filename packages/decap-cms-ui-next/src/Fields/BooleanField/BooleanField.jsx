import React from 'react';

import Field from '../../Field';
import Switch from '../../Switch';

function BooleanField({ name, onChange, children, value, ...props }) {
  return (
    <Field
      {...props}
      labelTarget={name}
      control
      onClick={() => (value ? onChange(false) : onChange(true))}
    >
      <Switch
        onChange={() => (value && onChange ? onChange(false) : onChange(true))}
        checked={value}
      />
      {children && children}
    </Field>
  );
}

export default BooleanField;
