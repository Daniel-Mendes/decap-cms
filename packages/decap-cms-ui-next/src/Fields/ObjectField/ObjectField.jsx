import React, { useState } from 'react';
import styled from '@emotion/styled';

import Field, { FieldContext } from '../../Field';
import Tree from '../../Tree';

const StyledField = styled(Field)`
  padding: 0 0 0 1rem;
`;

function ObjectField({ label, onChange, fields, filled, className }) {
  const [expanded, setExpanded] = useState(true);
  const [data, setData] = useState();
  const [treeType, setTreeType] = useState();

  function handleChange(changes) {
    const newData = { ...data, ...changes };
    setData(newData);
    onChange(newData);
  }

  return (
    <StyledField
      className={className}
      filled={filled}
      insideStyle={{ paddingTop: 0, paddingBottom: 0, paddingRight: 0 }}
    >
      <Tree
        single
        onExpandToggle={() => setExpanded(!expanded)}
        expanded={expanded}
        label={`${label}`}
        description={data && !!Object.keys(data).length && data[Object.keys(data)[0]]}
        type={treeType}
        onHeaderMouseEnter={() => setTreeType('primary')}
        onHeaderMouseLeave={() => setTreeType(null)}
      >
        <FieldContext.Provider value={{ filled: true }}>
          {fields && fields(handleChange)}
        </FieldContext.Provider>
      </Tree>
    </StyledField>
  );
}

export default ObjectField;
