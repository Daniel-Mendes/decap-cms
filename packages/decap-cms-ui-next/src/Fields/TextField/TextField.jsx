import React, { useState, forwardRef } from 'react';
import styled from '@emotion/styled';

import Field from '../../Field';

export const StyledInput = styled.input`
  color: ${({ theme }) => theme.color.highEmphasis};
  background: none;
  border: none;
  outline: none;
  width: calc(100% + 32px);
  font-family: inherit;
  font-size: ${({ title }) => (title ? '2rem' : '1rem')};
  font-weight: ${({ title }) => (title ? 'bold' : 'normal')};
  letter-spacing: ${({ title }) => (title ? '-0.5px' : '0')};
  line-height: 1rem;
  caret-color: ${({ theme, error }) =>
    error ? theme.color.danger['900'] : theme.color.primary['800']};
  margin: -2rem -1rem -1rem -1rem;
  padding: 2rem 1rem 1rem 1rem;
  ${({ clickable }) => (clickable ? `cursor: pointer;` : ``)} ::placeholder {
    color: ${({ theme }) => theme.color.disabled};
  }
`;

const TextField = forwardRef(
  (
    {
      name,
      label,
      icon,
      description,
      status,
      onChange,
      onClick,
      readOnly,
      placeholder,
      children,
      value,
      focus,
      title,
      type,
      password,
      className,
      inline,
      error,
      errors,
    },
    ref,
  ) => {
    const [inputFocus, setInputFocus] = useState();

    return (
      <Field
        onClick={onClick}
        label={label}
        labelTarget={name}
        description={description}
        status={status}
        focus={focus || inputFocus}
        className={className}
        inline={inline}
        error={error}
        errors={errors}
        icon={icon}
        clickable={!readOnly}
      >
        <StyledInput
          ref={ref}
          clickable={readOnly && !!onClick}
          onClick={onClick}
          readOnly={readOnly}
          type={password ? 'password' : type ? type : 'text'}
          id={name}
          name={name}
          placeholder={placeholder ? placeholder : label ? `Type ${label.toLowerCase()} here` : ''}
          value={value || ''}
          onChange={e => onChange(e)}
          onFocus={() => setInputFocus(true)}
          onBlur={() => setInputFocus(false)}
          autoComplete="off"
          title={title}
          error={error}
        />
        {children && children}
      </Field>
    );
  },
);

TextField.displayName = 'TextField';

export default TextField;
