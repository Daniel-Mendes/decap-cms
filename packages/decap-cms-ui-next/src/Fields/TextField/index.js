export { default, StyledInput } from './TextField';
