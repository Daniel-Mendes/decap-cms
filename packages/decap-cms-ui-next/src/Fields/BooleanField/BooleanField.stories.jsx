import React from 'react';
import styled from '@emotion/styled';
import { useArgs } from '@storybook/preview-api';

import BooleanField from '.';

const StyledBooleanField = styled(BooleanField)`
  width: 33vw;
`;

export default {
  title: 'Fields/BooleanField',
  component: BooleanField,
  argTypes: {
    errors: {
      control: 'object',
      if: {
        arg: 'error',
      },
    },
  },
  args: {
    label: 'Boolean Field Label',
    description: 'This is a description',
    status: 'Required',
    value: false,
    inline: false,
    error: false,
    errors: [
      {
        message: 'Error message.',
      },
    ],
  },
};

export function _BooleanField(args) {
  const [{ value }, updateArgs] = useArgs();

  function toggleValue() {
    updateArgs({ value: !value });
  }

  return <StyledBooleanField {...args} value={value} onChange={toggleValue} />;
}
