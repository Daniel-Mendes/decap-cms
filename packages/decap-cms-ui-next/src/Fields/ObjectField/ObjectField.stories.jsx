import React from 'react';
import styled from '@emotion/styled';
import { useArgs } from '@storybook/preview-api';

import ObjectField from '.';
import TextField from '../TextField';
import BooleanField from '../BooleanField';

const StyledObjectField = styled(ObjectField)`
  width: 33vw;
`;

export default {
  title: 'Fields/ObjectField',
  component: ObjectField,
  argTypes: {
    fields: {
      control: 'object',
    },
  },
  args: {
    name: 'links',
    label: 'Link',
    fields: {
      text: '',
      path: '',
      newWindow: false,
    },
    inline: false,
  },
};

export function _ObjectField(args) {
  const [{ fields }, updateArgs] = useArgs();

  function handleChange(fields) {
    updateArgs({ fields });
  }

  return (
    <StyledObjectField
      {...args}
      onChange={handleChange}
      fields={setObjectItemValue => (
        <React.Fragment>
          <TextField
            name="featureLinkText"
            label="Text"
            value={fields && fields.text}
            onChange={text => setObjectItemValue({ text })}
          />
          <TextField
            name="featureLinkPath"
            label="Path"
            value={fields && fields.path}
            onChange={path => setObjectItemValue({ path })}
          />
          <BooleanField
            name="newWindow"
            label="Open In New Window"
            value={fields && fields.newWindow}
            onChange={newWindow => setObjectItemValue({ newWindow })}
          />
        </React.Fragment>
      )}
    />
  );
}
