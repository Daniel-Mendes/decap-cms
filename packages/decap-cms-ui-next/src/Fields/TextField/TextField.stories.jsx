import React from 'react';
import styled from '@emotion/styled';
import { useArgs } from '@storybook/preview-api';

import TextField from '.';

const StyledTextField = styled(TextField)`
  width: 33vw;
`;

export default {
  title: 'Fields/TextField',
  component: TextField,
  argTypes: {
    errors: {
      control: 'object',
      if: {
        arg: 'error',
      },
    },
  },
  args: {
    label: 'Text Field Label',
    placeholder: 'Type something here',
    description: 'This is a description',
    status: 'Required',
    value: '',
    title: false,
    filled: false,
    error: false,
    errors: [
      {
        message: 'Error message.',
      },
    ],
  },
};

export function _TextField(args) {
  const [{ value }, updateArgs] = useArgs();

  function handleChange(e) {
    updateArgs({ value: e.target.value });
  }

  return <StyledTextField {...args} value={value} onChange={handleChange} />;
}
