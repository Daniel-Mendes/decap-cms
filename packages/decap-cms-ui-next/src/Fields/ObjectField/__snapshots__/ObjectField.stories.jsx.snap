// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields/ObjectField _ObjectField smoke-test 1`] = `"<div class="e1wdm66w0 e6zctmv0 css-1j1albh-FieldWrap-StyledField-StyledObjectField e1x0ipyd6"><div class="css-t9hsmw-FieldInside-FieldInside e1x0ipyd5" style="padding-top: 0px; padding-bottom: 0px; padding-right: 0px;"><label class="css-r9gw8d-Label-StyledLabel e1x0ipyd3"> </label><div class="css-dj548v-ChildrenWrap e1x0ipyd4"><div class="css-6et2o-TreeWrap e1vyjth0"><div class="css-1fe3spy-TreeHeader e1vyjth5"><button type="primary" class="e1vyjth9 css-1w966m3-StyledButton-StyledButton-ExpandButton e9pg3mh2" aria-expanded="true" data-no-dnd="true"><svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" class="css-11wxfmo-ExpandIcon e1vyjth8" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><polyline points="9 18 15 12 9 6"></polyline></svg></button><div class="css-wjjdoj-TreeHeaderText e1vyjth4"><label class="css-1wcy17e-Label-StyledLabel e1vyjth7">Link</label></div></div><div class="css-4j3v1g-TreeContentWrap e1vyjth2" style="height: auto; margin-top: -0.5rem;"><div data-no-dnd="true" class="css-9d6ng3-TreeContent e1vyjth1" style="transform: translate3d(0px, 0px, 0px); opacity: 1;"><div class="css-1rig8r-FieldWrap e1x0ipyd6"><div class="css-1mlik5w-FieldInside-FieldInside-FieldInside-FieldInside-FieldInside e1x0ipyd5"><label for="featureLinkText" class="css-11m1vnj-Label-StyledLabel e1x0ipyd3">Text </label><div class="css-dj548v-ChildrenWrap e1x0ipyd4"><input type="text" id="featureLinkText" name="featureLinkText" placeholder="Type text here" autocomplete="off" class="css-h0pblb-StyledInput e4jt8f0" value=""></div></div></div><div class="css-1rig8r-FieldWrap e1x0ipyd6"><div class="css-1mlik5w-FieldInside-FieldInside-FieldInside-FieldInside-FieldInside e1x0ipyd5"><label for="featureLinkPath" class="css-11m1vnj-Label-StyledLabel e1x0ipyd3">Path </label><div class="css-dj548v-ChildrenWrap e1x0ipyd4"><input type="text" id="featureLinkPath" name="featureLinkPath" placeholder="Type path here" autocomplete="off" class="css-h0pblb-StyledInput e4jt8f0" value=""></div></div></div><div class="css-1rig8r-FieldWrap e1x0ipyd6"><div class="css-1mlik5w-FieldInside-FieldInside-FieldInside-FieldInside-FieldInside e1x0ipyd5"><label for="newWindow" class="css-11m1vnj-Label-StyledLabel e1x0ipyd3">Open In New Window </label><div class="css-dj548v-ChildrenWrap e1x0ipyd4"><input type="checkbox" class="css-1rg9wi1-StyledInput e1lye9la1"><button class="css-1407tdr-SwitchButton e1lye9la0"></button></div></div></div></div></div></div></div></div></div>"`;
